src/core/deltafilewrapper.cpp

```diff
@@ -473,24 +473,73 @@ void DeltaFileWrapper::addPatch( const QString &localLayerId, const QString &sou
     areFeaturesEqual = true;
   }
 
-  // TODO types should be checked too, however QgsFields::operator== is checking instances, not values
-  // TODO be careful with calculated fields here!!! Needs a fix!
-  Q_ASSERT( oldFeature.fields().names() == newFeature.fields().names() );
+  QgsFields fields;
+  QgsFields oldFields = oldFeature.fields();
+  QgsFields newFields = newFeature.fields();
+  int ignoredFields = 0;
+
+  for ( int idx = 0, l = newFields.count(); idx < l; idx++ )
```

Comment:
```suggestion
  const int fieldCount = newFields.count();
  for ( int idx = 0; idx < fieldCount; idx++ )
```

Would that be appropriate? Easier to read?

Reply:
I did it with `idx`, since I need to work with it in the loop, otherwise I need to obtain/store the current index in another way.

Reply:
Is `l` modified?

Reply:
sorry, misread your suggestion in the morning.